src/main/knn/KnnGraphTester.java

```diff
@@ -731,71 +755,22 @@ private int[][] computeNN(Path docPath, Path queryPath, VectorEncoding encoding)
     return result;
   }
 
-  private int createIndex(Path docsPath, Path indexPath) throws IOException {
-    IndexWriterConfig iwc = new IndexWriterConfig().setOpenMode(IndexWriterConfig.OpenMode.CREATE);
-    iwc.setCodec(getCodec(maxConn, beamWidth, exec, numMergeWorker, quantize));
-    // iwc.setMergePolicy(NoMergePolicy.INSTANCE);
-    iwc.setRAMBufferSizeMB(WRITER_BUFFER_MB);
-    iwc.setUseCompoundFile(false);
-
-    FieldType fieldType =
-        switch (vectorEncoding) {
-          case BYTE -> KnnByteVectorField.createFieldType(dim, similarityFunction);
-          case FLOAT32 -> KnnFloatVectorField.createFieldType(dim, similarityFunction);
-        };
-    if (quiet == false) {
-      iwc.setInfoStream(new PrintStreamInfoStream(System.out));
-      System.out.println("creating index in " + indexPath);
-    }
-    long start = System.nanoTime();
-    try (FSDirectory dir = FSDirectory.open(indexPath);
-         IndexWriter iw = new IndexWriter(dir, iwc)) {
-      try (FileChannel in = FileChannel.open(docsPath)) {
-        VectorReader vectorReader = VectorReader.create(in, dim, vectorEncoding);
-        for (int i = 0; i < numDocs; i++) {
-          Document doc = new Document();
-          switch (vectorEncoding) {
-            case BYTE -> doc.add(
-                new KnnByteVectorField(
-                    KNN_FIELD, ((VectorReaderByte) vectorReader).nextBytes(), fieldType));
-            case FLOAT32 -> doc.add(
-                new KnnFloatVectorField(KNN_FIELD, vectorReader.next(), fieldType));
-          }
-          doc.add(new StoredField(ID_FIELD, i));
-          iw.addDocument(doc);
-          if (randomCommits && i % 500 == 0 && Math.random() < 0.5) {
-            iw.commit();
-          }
-        }
-        if (quiet == false) {
-          System.out.println("Done indexing " + numDocs + " documents; now flush");
-        }
-      }
-    }
-    long elapsed = System.nanoTime() - start;
-    if (quiet == false) {
-      System.out.println(
-          "Indexed " + numDocs + " documents in " + TimeUnit.NANOSECONDS.toSeconds(elapsed) + "s");
-    }
-    return (int) TimeUnit.NANOSECONDS.toMillis(elapsed);
-  }
-
-  private static Codec getCodec(int maxConn, int beamWidth, ExecutorService exec, int numMergeWorker, boolean quantize) {
+  private static Codec getCodec(int maxConn, int beamWidth, ExecutorService exec, int numMergeWorker, boolean quantize, int quantizeBits, boolean quantizeCompress) {
     if (exec == null) {
       return new Lucene99Codec() {
         @Override
         public KnnVectorsFormat getKnnVectorsFormatForField(String field) {
           return quantize ?
-              new Lucene99HnswScalarQuantizedVectorsFormat(maxConn, beamWidth) :
-              new Lucene99HnswVectorsFormat(maxConn, beamWidth);
+              new Lucene99HnswScalarQuantizedVectorsFormat(maxConn, beamWidth, numMergeWorker, quantizeBits, quantizeCompress, null, null) :
```

Comment:
strange I swear I checked that this compiled and I was able to run it at least once. Maybe I ran against an older Lucene checkout??